### PR TITLE
feat: my settings page tests

### PR DESCRIPTION
**Pull request by ahmadshaheer**
### Summary
Test cases:
- [x] Should display Dark and Light theme buttons properly
- [x] Should activate Dark and Light buttons on click
- [x] Should switch the them on clicking Light theme
- [x] Should properly display the User Details Form
- [x] Should update the name on clicking Update button
- [ ] Should redirect to /login on clicking Logout
- [x] Should properly display the Password Reset Form
- [x] Should display validation error if password is less than 8 characters
- [x] Should check if the "Change Password" button is disabled in case current / new password is less than 8 characters

#### Related Issues / PR's
close https://github.com/SigNoz/signoz/issues/5501

#### Screenshots

NA

### CODE REVIEW

File: frontend/src/mocks-server/handlers.ts [hunk 1]
```diff
@@ -97,4 +97,23 @@ export const handlers = [
 	rest.post('http://localhost/api/v1/invite', (_, res, ctx) =>
 		res(ctx.status(200), ctx.json(inviteUser)),
 	),
+	rest.put('http://localhost/api/v1/user/:id', (_, res, ctx) => {
+		console.log({ _ });
```
Comment (SagarRajput-7) [inline frontend/src/mocks-server/handlers.ts:101]: remove console.log

File: frontend/src/container/MySettings/__tests__/MySettings.test.tsx [hunk 2]
```diff
@@ -0,0 +1,236 @@
+// import ROUTES from 'constants/routes';
+import MySettingsContainer from 'container/MySettings';
+import { act, fireEvent, render, screen, waitFor } from 'tests/test-utils';
+
+// const historyPush = jest.fn();
+// jest.mock('lib/history', () => ({
+// 	__esModule: true,
+// 	push: historyPush,
+// }));
```
Comment (SagarRajput-7) [inline frontend/src/container/MySettings/__tests__/MySettings.test.tsx:9]: remove commented code

File: frontend/src/container/MySettings/__tests__/MySettings.test.tsx [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+	// describe('Logout button', () => {
+	// 	it('Should redirect to /login on clicking Logout', async () => {
+	// 		const logoutButton = screen.getByTestId('logout-button');
+	// 		fireEvent.click(logoutButton);
+	// 		await waitFor(() => {
+	// 			expect(historyPush).toHaveBeenCalledWith(ROUTES.LOGIN);
+	// 		});
+	// 	});
+	// });
```
Comment (SagarRajput-7) [inline frontend/src/container/MySettings/__tests__/MySettings.test.tsx:136]: remove commented code

File: frontend/src/container/MySettings/__tests__/MySettings.test.tsx (hunk 2 above)
Comment (ahmadshaheer) [reply frontend/src/container/MySettings/__tests__/MySettings.test.tsx:9]: Done

File: frontend/src/mocks-server/handlers.ts (hunk 1 above)
Comment (ahmadshaheer) [reply frontend/src/mocks-server/handlers.ts:101]: Done

File: frontend/src/container/MySettings/__tests__/MySettings.test.tsx [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+		});
+
+		it('Should switch the them on clicking Light theme', async () => {
+			const lightThemeOption = screen.getByRole('radio', {
+				name: /light/i,
+			});
+			fireEvent.click(lightThemeOption);
+
+			await waitFor(() => {
+				expect(toggleThemeFunction).toBeCalled();
```
Comment (SagarRajput-7) [inline frontend/src/container/MySettings/__tests__/MySettings.test.tsx:82]: NIT: could also check - `calledWith` value

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/container/MySettings/__tests__/MySettings.test.tsx (hunk 4 above)
Comment (vikrantgupta25) [reply frontend/src/container/MySettings/__tests__/MySettings.test.tsx:82]: it is not called with anything, it reverses the global state

**Review by vikrantgupta25 — APPROVED** [review]
please take care of the comments. rest LGTM

File: frontend/src/container/MySettings/__tests__/MySettings.test.tsx (hunk 3 above)
Comment (ahmadshaheer) [reply frontend/src/container/MySettings/__tests__/MySettings.test.tsx:136]: Done

File: frontend/src/container/MySettings/__tests__/MySettings.test.tsx (hunk 2 above)
Comment (ahmadshaheer) [reply frontend/src/container/MySettings/__tests__/MySettings.test.tsx:9]: Done

File: frontend/src/mocks-server/handlers.ts (hunk 1 above)
Comment (ahmadshaheer) [reply frontend/src/mocks-server/handlers.ts:101]: Done

**Comment by SagarRajput-7** [comment]
@ahmadshaheer - build is failing please check - 

```
`/home/runner/work/signoz/signoz/frontend/src/mocks-server/handlers.ts
  167:5  error  Parsing error: ',' expected

/home/runner/work/signoz/signoz/frontend/src/mocks-server/server.ts
  4:26  error  Parse errors in imported module './handlers': ',' expected. (167:5)  import/namespace`
```

**Review by SagarRajput-7 — APPROVED** [review]

**Review by vikrantgupta25 — APPROVED** [review]

**Pull request merged**